Client/Frontend/Shields/ShieldsViewController.swift

```diff
@@ -70,12 +70,12 @@ class ShieldsViewController: UIViewController, PopoverContentComponent {
 //        shieldsView.shieldsContainerStackView.fingerprintingStatView.valueLabel.text = String(shieldBlockStats.fingerprintCount)
     }
     
-    private func updateBraveShieldState(shield: BraveShieldState.Shield, on: Bool) {
+    private func updateBraveShieldState(shield: BraveShieldState.Shield, on: Bool, option: Preferences.Option<Bool>?) {
         guard let url = tab.url else { return }
         let allOff = shield == .AllOff
         // `.AllOff` uses inverse logic
         // Technically we set "all off" when the switch is OFF, unlike all the others
-        let isOn = allOff ? !on : on
+        let isOn = allOff ? !on : (on == option?.value ? nil : on)
```

Comment:
Can you add a comment explaining this a bit. Totally understand it, just want the "why" added here too.